### PR TITLE
add layout+yolov8

### PR DESCRIPTION
**Pull request by chengtianwu**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/layout/yolov8/README_CN.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+# YOLOv8
+<!--- Guideline: use url linked to abstract in ArXiv instead of PDF for fast loading.  -->
+
+> [YOLOv8: You Only Look Once Version 8](https://github.com/ultralytics/ultralytics)
+
+## 1. 模型描述
+<!--- Guideline: Introduce the model and architectures. Cite if you use/adopt paper explanation from others. -->
+
+YOLOv8 是Ultralytics的YOLO的最新版本。作为一种前沿、最先进(SOTA)的模型，YOLOv8在之前版本的成功基础上引入了新功能和改进，以提高性能、灵活性和效率。YOLOv8支持全范围的视觉AI任务，包括检测, 分割, 姿态估计, 跟踪, 和分类。这种多功能性使用户能够利用YOLOv8的功能应对多种应用和领域的需求。
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:11]: 已修改

File: configs/layout/yolov8/README_CN.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+<div align="center">
+
+| **模型** | **环境配置** | **平均准确率（mAP）** | **训练时间** | **FPS** | **配置文件** | **模型权重下载** |
+| :-----: | :-----:  | :-----: | :-----: | :-----: |:--------: | :-----: |
+| YOLOv8 | D910x2-MS2.2-G | 94.4% | 420 ms/epoch | 38.02 img/s | [yaml](https://github.com/mindspore-lab/mindocr/blob/main/configs/layout/yolov8/yolov8n.yaml) | [ckpt](https://download.mindspore.cn/toolkits/mindocr/yolov8/yolov8n-4b9e8004.ckpt) \| [mindir](https://download.mindspore.cn/toolkits/mindocr/yolov8/yolov8n-2a1f68ab.mindir) |
+</div>
+
+**注意:**
+
+- 环境配置：训练的环境配置表示为 {处理器}x{处理器数量}-{MS模式}，其中 Mindspore 模式可以是 G-graph 模式或 F-pynative 模式。例如，D910x8-MS1.10-G 用于使用图形模式在4张昇腾910 NPU上依赖Mindspore1.10版本进行训练。
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:40]: 1.已修改
2.已修改
3.和其他README风格一致

File: configs/layout/yolov8/README_CN.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+```
+
+下载完成后，可以使用上述MindOCR提供的脚本将数据转换为YOLOv8输入格式的数据类型。
+
+#### 3.1.3 检查配置文件
+除了数据集的设置，请同时重点关注以下变量的配置：`system.distribute`, `system.val_while_train`, `common.batch_size`, `train.ckpt_save_dir`, `train.dataset.dataset_path`, `eval.ckpt_load_path`, `eval.dataset.dataset_path`, `eval.loader.batch_size`。说明如下：
+
+```yaml
+system:
+  distribute: &distribute True                                    # 分布式训练为True，单卡训练为False
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:71]: 已修改

File: configs/layout/yolov8/README_CN.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+  amp_level: 'O0'
+  amp_level_infer: "O0"
+  seed: 42
+  val_while_train: False                                               # 边训练边验证
+  drop_overflow_update: False
+common:
+  ...
+  batch_size: &batch_size 16                                         # 训练批大小
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:79]: 已修改

File: mindocr/models/backbones/_registry.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-'''backbone registry and list'''
+"""backbone registry and list"""
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply mindocr/models/backbones/_registry.py:1]: 文档注释用双引号，其他地方是字符串可以用单引号，也可以都用双引号。
统一使用双引号，已修改其他地方

File: tools/train.py [hunk 6]
```diff
@@ -107,12 +107,6 @@ def main(cfg):
 
     # create model
     amp_level = cfg.system.get("amp_level", "O0")
-    if ms.get_context("device_target") == "GPU" and cfg.system.val_while_train and amp_level == "O3":
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply tools/train.py:110]: 已修改

File: tools/eval.py [hunk 7]
```diff
@@ -144,7 +144,7 @@ def main(cfg):
 
 
 if __name__ == "__main__":
-    # argpaser
+    # args parser
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply tools/eval.py:147]: 已修改

File: deploy/py_infer/src/data_process/preprocess/transforms/layout_transforms.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
+import numpy as np
+import cv2
+
+
+def letterbox(scaleup):
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply deploy/py_infer/src/data_process/preprocess/transforms/layout_transforms.py:5]: 1. 不冗余，如果不封装会和其他数据增强方式调用方式差别较大。pipeline通过以下方式调用：
    ops = letterbox
    ops = ops(**params)
    data = ops(data)
2. 装饰器主要用来减少冗余代码的，重复代码只有func这一段，用了装饰器意义不是太大。

File: deploy/py_infer/src/data_process/preprocess/transforms/layout_transforms.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        r = min(new_shape[0] / shape[0], new_shape[1] / shape[1])
+        if not scaleup:  # only scale down, do not scale up (for better test mAP)
+            r = min(r, 1.0)
+
+        # Compute padding
+        new_unpad = int(round(shape[1] * r)), int(round(shape[0] * r))
+        dw, dh = new_shape[1] - new_unpad[0], new_shape[0] - new_unpad[1]  # wh padding
+
+        dw /= 2  # divide padding into 2 sides
+        dh /= 2
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply deploy/py_infer/src/data_process/preprocess/transforms/layout_transforms.py:29]: 已修改

File: configs/layout/yolov8/yolov8n.yaml [hunk 10]
```diff
@@ -0,0 +1,153 @@
+system:
+  mode: 0 # 0 for graph mode, 1 for pynative mode in MindSpore
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/yolov8n.yaml:2]: 已修改当前commit，其他模型的yaml有同样的问题，一共36个。另提PR改。

File: configs/layout/yolov8/README_CN.md [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+
+可以使用以下代码将推理结果进行可视化：
+
+```python
+from matplotlib import pyplot as plt
+import matplotlib.patches as patches
+from PIL import Image
+
+
+img = Image.open('publaynet/val/PMC4958442_00003.jpg')
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:183]: 已修改

File: configs/layout/yolov8/README_CN.md [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+python infer.py \
+    --input_images_dir=/your_path_to/val \
+    --layout_model_path=your_path_to/output.mindir \
+    --layout_model_name_or_config=../../configs/layout/yolov8/yolov8n.yaml \
+    --layout_save_dir=results_dir
+```
+
+## 6. 可视化
+
+可以使用以下代码将推理结果进行可视化：
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:175]: 当前MindOCR已有可视化模块，把这段代码融进去略复杂，而且现在可视化目录和逻辑略混乱，需要重构。后续修改

File: configs/layout/yolov8/README_CN.md [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+    dataset_path: publaynet/val.txt                           # 验证数据集路径
+  ...
+  loader:
+    shuffle: False
+    batch_size: 16                                                 # 验证批大小
+...
+```
+
+**注意:**
+- 由于全局批大小 （batch_size x num_devices） 是对结果复现很重要，因此当GPU/NPU卡数发生变化时，调整`batch_size`以保持全局批大小不变，或根据新的全局批大小线性调整学习率。
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:103]: 已修改当前commit。其他readme有同样问题。后续对整个文档进行重构时修复。

File: configs/layout/yolov8/README_CN.md [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+# YOLOv8
+<!--- Guideline: use url linked to abstract in ArXiv instead of PDF for fast loading.  -->
+
+> [YOLOv8: You Only Look Once Version 8](https://github.com/ultralytics/ultralytics)
+
+## 1. 模型描述
+<!--- Guideline: Introduce the model and architectures. Cite if you use/adopt paper explanation from others. -->
+
+YOLOv8 是Ultralytics的YOLO的最新版本。作为一种前沿、最先进(SOTA)的模型，YOLOv8在之前版本的成功基础上引入了新功能和改进，以提高性能、灵活性和效率。YOLOv8支持全范围的视觉AI任务，包括检测, 分割, 姿态估计, 跟踪, 和分类。这种多功能性使用户能够利用YOLOv8的功能应对多种应用和领域的需求。
+
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md:12]: 已修改

File: configs/layout/yolov8/README_CN.md [hunk 15]
```diff
@@ -0,0 +1,232 @@
...
+<!--- Guideline: If an architecture table/figure is available in the paper, put one here and cite for intuitive illustration. -->
+
+![](images/yolov8.png)
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README_CN.md]: 已修改

File: configs/layout/yolov8/README.md [hunk 16]
```diff
@@ -0,0 +1,226 @@
+English | [中文](https://github.com/mindspore-lab/mindocr/blob/main/configs/layout/yolov8/README_CN.md)
```
Comment (tonytonglt) [inline configs/layout/yolov8/README.md:1]: 中文链接404

File: configs/layout/yolov8/README.md [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+### 3.1 Preparation
+
+#### 3.1.1 Installation
+Please refer to the [installation instruction](https://github.com/mindspore-lab/mindocr#installation) in MindOCR.
+
+#### 3.1.2 PubLayNet Dataset Preparation
+
+PubLayNet is a dataset for document layout analysis. It contains images of research papers and articles and annotations for various elements in a page such as "text", "list", "figure" etc in these research paper images. The dataset was obtained by automatically matching the XML representations and the content of over 1 million PDF articles that are publicly available on PubMed Central.
+
+#### 3.1.4 Check YAML Config Files
```
Comment (tonytonglt) [inline configs/layout/yolov8/README.md:59]: 3.1.3？

File: configs/layout/yolov8/README.md [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+- Download: url of the pretrained model weights. Use absolute url path.
+-->
+
+### Accuracy
+
+According to our experiment, the evaluation results on the public benchmark dataset (PublayNet) are as follows:
+
+<div align="center">
+
+| **Model** | **Context** | **Mean Average accuracy (mAP)** | **Train T.** | **FPS** | **Recipe** | **Download** |
```
Comment (tonytonglt) [inline configs/layout/yolov8/README.md:37]: 最好加上per step time，单位ms/step

File: configs/layout/yolov8/README.md (hunk 16 above)
Comment (chengtianwu) [reply configs/layout/yolov8/README.md:1]: 要合进去才能有链接

File: configs/layout/yolov8/README.md (hunk 17 above)
Comment (chengtianwu) [reply configs/layout/yolov8/README.md:59]: 已修改

**Review by tonytonglt — APPROVED** [review]

File: configs/layout/yolov8/README.md [hunk 19]
```diff
@@ -0,0 +1,226 @@
...
+| :-----: | :-----------: | :--------------: | :----------: | :--------: | :--------: |:----------: |
+| YOLOv8 | D910x2-MS2.2-G | 94.4% | 420 ms/epoch | 38.02 img/s | [yaml](https://github.com/mindspore-lab/mindocr/blob/main/configs/layout/yolov8/yolov8n.yaml) | [ckpt](https://download.mindspore.cn/toolkits/mindocr/yolov8/yolov8n-4b9e8004.ckpt) \| [mindir](https://download.mindspore.cn/toolkits/mindocr/yolov8/yolov8n-2a1f68ab.mindir) |
+</div>
+
+**Notes:**
+- Context: Training context denoted as {device}x{pieces}-{MS mode}, where mindspore mode can be G-graph mode or F-pynative mode with ms function. For example, D910x4-MS2.2-G is for training on 4 pieces of Ascend 910 NPU using graph mode based on MindSpore version 2.2.
+- To reproduce the result on other contexts, please ensure the global batch size is the same.
+- The models are trained from scratch without any pre-training. For more dataset details of training and evaluation, please refer to [Dataset Download & Dataset Usage](#312-dataset-download) section.
```
[the remark this answers is not in the document]
Comment (chengtianwu) [reply configs/layout/yolov8/README.md:45]: 已修改

File: configs/layout/yolov8/README.md (hunk 18 above)
Comment (chengtianwu) [reply configs/layout/yolov8/README.md:37]: 就是per step time，写错了，已修改

**Review by panshaowu — APPROVED** [review]
